### PR TITLE
Less files for environment variables management

### PR DESCRIPTION
**Pull request by JuanitoFatas**
This Pull Request cleans up environment variable management. If this PR merged, one only needs to clone this PR and good to go! Ever need to override anything, put in `.env.local`. Test specific? Put in `.env.test`. :)

Ref. https://github.com/bkeepers/dotenv#multiple-rails-environments

<details>
  <summary>Details</summary>
- commit `.env` so faster developer setup
- ignore `.env.local`
- create failed environment variable in `.env` to tell something needs to
override, e.g.
    `INTERCOM_API_SECRET=intercom_api_secret`
- if needs to override, put in `.env.local`
- remove unnecessary files
- remove unnecessary lines in `spec/spec_helper.rb`
- `.env.{test,development}` is for environment-specific ENV
</details>

### CODE REVIEW

File: .env.test.sample [hunk 1]
```diff
@@ -1,8 +0,0 @@
-SOURCE_URL="https://status.example.com/api/endpoint"
-DURATION="duration"
-STARTED_AT="started_at"
-INCIDENT_ID="incident_id"
-STATE="state"
-RESOLVED_AT="resolved_at"
-REVIEW="review"
-FOLLOWUP_DAYS="5"
```
Comment (joshuatobin) [inline .env.test.sample]: I'm not sure we want to be removing this file.

File: .gitignore [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
 /tmp
 /config/database.yml
-/.env
 /public/assets
-
 log
-
 *.swp
-.env
-.env.test
```
Comment (joshuatobin) [inline .gitignore]: Same here. I think we want to have a .gitignore for .env.test

Comment (reidmix) [inline .gitignore]: Agreed, and `.env`

File: .travis.yml [hunk 3]
```diff
@@ -4,7 +4,6 @@ rvm:
 sudo: false
 bundler_args: --without development
 before_script:
-  - cp .env.test{.sample,} && cp .env{.sample,}
```
Comment (reidmix) [inline .travis.yml]: I think we can keep this line too (maybe just remove the `.env` copy)

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,9 +1,6 @@
 ENV["RAILS_ENV"] ||= 'test'
 ENV["HOSTNAME"] = 'retrodot.test'
 
-require 'dotenv'
-Dotenv.load!('.env.test')
-
```
Comment (reidmix) [inline spec/spec_helper.rb]: Ditto, keep this.

**Comment by reidmix** [comment]
I think there is confusion I can help with 💃 We use:
- `.env.development` is for when `RAILS_ENV=development` and is typically for when running your local server or pry, etc.
- `.env.test` is for `RAILS_ENV=test` and used by rspec and has test envvars needed for the specs to run 
- `.env` and `.env.local` is for all `RAILS_ENV`s

**Comment by JuanitoFatas** [comment]
They have exact same contents:

- https://github.com/heroku/retrodot/blob/master/.env.test.sample
- https://github.com/heroku/retrodot/blob/master/.env.development.sample

So we should put into .env right? :)

.env will be loaded before the environment-specific env variants (.env.test, .env.development):

> When you load a certain environment, dotenv will first load general env vars from .env, then load environment specific env vars from .env.<current environment>. Variables defined in .env.<current environment> will override any values set in .env or already defined in the environment.
> -- https://github.com/bkeepers/dotenv#multiple-rails-environments

**Comment by reidmix** [comment]
Good point.  Maybe?  @joshuatobin i know your usecase was that `.env` and `.env.test` were different and one leaking into the other?  I forget the specifics.

File: .env.test.sample (hunk 1 above)
Comment (jmervine) [inline .env.test.sample]: @JuanitoFatas (cc @joshuatobin, @lexelby ) We tend to standardize on using `.env.test` to bootstrap our test configurations. As such, we'll probably want to keep this laying around. The one recommendation I can make is that we may want to forego having a `.env.test.sample` all together, in favor of just having `.env.test`. `.sample` indicates that we're expecting the developer to change something before using. In the case of this file, we'll want them to just use what's there in most cases.

File: .gitignore (hunk 2 above)
Comment (jmervine) [inline .gitignore]: @JuanitoFatas (cc @joshuatobin, @lexelby, @reidmix) Per my comment above, we may want to leave `.env.test` off the ignore list. However, I strongly agree the `.env` should be in this file.

File: .travis.yml (hunk 3 above)
Comment (jmervine) [inline .travis.yml]: @JuanitoFatas (cc @joshuatobin, @lexelby, @reidmix) 👍 on this comment, also, once again, see my comment RE `.env.test` above.

**Comment by JuanitoFatas** [comment]
Per https://github.com/heroku/retrodot/pull/29#discussion_r76827655, let's keep these envs as is. :) Thanks for the feedback!

**Comment by joshuatobin** [comment]
@JuanitoFatas <3 - thanks for all the work thus far.

**Pull request closed without merging**